### PR TITLE
MUL-211 - Primary order hiding

### PR DESCRIPTION
**Pull request by MacBalc**

### CODE REVIEW

File: src/Repository/OrderRepository.php [hunk 1]
```diff
@@ -61,10 +74,112 @@ public function createByCustomerAndChannelIdAndSecondaryQueryBuilder(int $custom
...
             ->andWhere('o.state != :state')
-            ->andWhere('o.primaryOrder is NOT NULL')
+            ->andWhere('o.mode = :secondaryOrderMode')
+            ->setParameter('secondaryOrderMode', OrderInterface::SECONDARY_ORDER_MODE)
             ->setParameter('customerId', $customerId)
             ->setParameter('channelId', $channelId)
             ->setParameter('state', OrderInterfaceAlias::STATE_CART)
             ;
     }
+
```
Comment (senghe) [inline src/Repository/OrderRepository.php:84]: The methods below are not related to the production code. I would create separate repository for this on `tests/Behat` namespace.

File: src/Repository/OrderRepository.php [hunk 2]
```diff
@@ -61,10 +74,112 @@ public function createByCustomerAndChannelIdAndSecondaryQueryBuilder(int $custom
             ->andWhere('o.customer = :customerId')
             ->andWhere('o.channel = :channelId')
             ->andWhere('o.state != :state')
-            ->andWhere('o.primaryOrder is NOT NULL')
+            ->andWhere('o.mode = :secondaryOrderMode')
```
Comment (senghe) [inline src/Repository/OrderRepository.php:77]: Do we have any test for this?

File: src/Repository/OrderRepository.php [hunk 3]
```diff
@@ -29,6 +31,17 @@ public function findAllByVendor(VendorInterface $vendor): QueryBuilder
             ;
     }
 
+    public function findAllSecondaryOrders(): QueryBuilder
```
Comment (senghe) [inline src/Repository/OrderRepository.php:34]: Do we have any test for this?

File: features/admin/customer_orders.feature [hunk 4]
```diff
@@ -0,0 +1,26 @@
+@hiding_primary_orders_in_customer_tab
+Feature: Hiding primary orders in customers tab
+  As an administrator
+  viewing the list of orders for a specific customer
```
Comment (senghe) [inline features/admin/customer_orders.feature:4]:
```suggestion
  During orders list view for a specific customer
```

File: features/shop/order.feature [hunk 5]
```diff
@@ -13,36 +13,32 @@ Feature: Spliting orders when cart was filled with products from different Vendo
...
+
+  @ui
+  Scenario: Browsing orders, admin cannot see primary orders
+    Given store has 4 products from different Vendors
+    And I have 3 products in cart
+    And I finalize order
+    Given I am logged in as an administrator
+    And I am on "/admin"
+    And I follow "Orders"
+    Then I should see 3 orders
```
Comment (senghe) [inline features/shop/order.feature:44]: I would put the order numbers as parameter. to prove the orders are correct.

File: src/Repository/OrderRepository.php [hunk 6]
```diff
@@ -61,10 +74,112 @@ public function createByCustomerAndChannelIdAndSecondaryQueryBuilder(int $custom
...
+            ->setParameter('state', OrderPaymentStates::STATE_PAID)
+            ->setParameter('startDate', $startDate)
+            ->setParameter('endDate', $endDate)
+            ->setParameter('mode', OrderInterface::PRIMARY_ORDER_MODE)
+            ->getQuery()
+            ->getSingleScalarResult()
+            ;
+    }
+
+    public function findLatestInChannel(int $count, ChannelInterface $channel): array
```
Comment (senghe) [inline src/Repository/OrderRepository.php:159]: Unused method

File: src/Repository/OrderRepository.php [hunk 7]
```diff
@@ -61,10 +74,112 @@ public function createByCustomerAndChannelIdAndSecondaryQueryBuilder(int $custom
...
+            ->setParameter('state', OrderPaymentStates::STATE_PAID)
+            ->setParameter('startDate', $startDate)
+            ->setParameter('endDate', $endDate)
+            ->setParameter('mode', OrderInterface::PRIMARY_ORDER_MODE)
+            ->getQuery()
+            ->getSingleScalarResult()
+            ;
+    }
+
+    public function countPaidByChannel(ChannelInterface $channel): int
```
Comment (senghe) [inline src/Repository/OrderRepository.php:122]: Unused method

File: src/Repository/OrderRepository.php [hunk 8]
```diff
@@ -61,10 +74,112 @@ public function createByCustomerAndChannelIdAndSecondaryQueryBuilder(int $custom
...
+            ->setParameter('state', \Sylius\Component\Core\Model\OrderInterface::STATE_CART)
+            ->setParameter('mode', OrderInterface::PRIMARY_ORDER_MODE)
+            ->setMaxResults($count)
+            ->getQuery()
+            ->getResult()
+            ;
+    }
+
+    /** @phpstan-ignore-next-line */
+    public function createByCustomerIdQueryBuilder($customerId): QueryBuilder
```
Comment (senghe) [inline src/Repository/OrderRepository.php:176]: Unused method

File: src/Repository/OrderRepository.php [hunk 9]
```diff
@@ -61,10 +74,112 @@ public function createByCustomerAndChannelIdAndSecondaryQueryBuilder(int $custom
...
+            ->setParameter('channel', $channel)
+            ->setParameter('state', \Sylius\Component\Core\Model\OrderInterface::STATE_CART)
+            ->setParameter('mode', OrderInterface::PRIMARY_ORDER_MODE)
+            ->setMaxResults($count)
+            ->getQuery()
+            ->getResult()
+            ;
+    }
+
+    /** @phpstan-ignore-next-line */
```
Comment (senghe) [inline src/Repository/OrderRepository.php:175]: Additionally, when you make line-ignore, please tell us, why you do it (in a comment)

File: src/Repository/PaymentRepository.php [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace BitBag\OpenMarketplace\Repository;
+
+use BitBag\OpenMarketplace\Entity\OrderInterface;
+use Doctrine\ORM\QueryBuilder;
+use Sylius\Bundle\CoreBundle\Doctrine\ORM\PaymentRepository as BasePaymentRepository;
+
+class PaymentRepository extends BasePaymentRepository
+{
+    public function findPaymentsOfSecondaryOrders(): QueryBuilder
```
Comment (senghe) [inline src/Repository/PaymentRepository.php:20]: Incorrect method name. It does not find. It returns query builder for secondary order payments.

File: features/admin/payment_viewing.feature [hunk 11]
```diff
@@ -0,0 +1,24 @@
+@payment_viewing
+Feature: Login in to admin panel
+  and going to payments tab
+  i should not see primary order payments
```
Comment (senghe) [inline features/admin/payment_viewing.feature:4]:
```suggestion
  I should not see primary order payments
```

File: spec/NumberAssigner/OrderNumberAssignerSpec.php [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    public function it_is_initializable(): void
+    {
+        $this->shouldHaveType(OrderNumberAssigner::class);
+    }
+
+    public function it_implements_interface(): void
+    {
+        $this->shouldImplement(\BitBag\OpenMarketplace\NumberAssigner\OrderNumberAssignerInterface::class);
```
Comment (senghe) [inline spec/NumberAssigner/OrderNumberAssignerSpec.php:37]: Missing import statement

File: src/Repository/ShipmentRepository.php [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace BitBag\OpenMarketplace\Repository;
+
+use BitBag\OpenMarketplace\Entity\OrderInterface;
+use Doctrine\ORM\QueryBuilder;
+use Sylius\Bundle\CoreBundle\Doctrine\ORM\ShipmentRepository as BasePaymentRepository;
+
+class ShipmentRepository extends BasePaymentRepository
+{
+    public function findShipmentsWithoutPrimaryModeOrder(): QueryBuilder
```
Comment (senghe) [inline src/Repository/ShipmentRepository.php:20]: Incorrect method name

File: features/admin/customer_orders.feature [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+    And the store has "UPS" shipping method with "$20.00" fee per unit for "Web-EU" channel
+    And the store has also a payment method "Bank transfer" with a code "transfer"
+    And store has primary and secondary order with payment state "paid"
+
+  @ui
+  Scenario: Viewing sales summary
+    Given I am on "/admin"
+    And I follow "Customers"
+    And I follow "Show orders"
+    Then I should see 1 orders
```
Comment (senghe) [inline features/admin/customer_orders.feature:26]: Where is the information, it's not the primary order?

File: features/admin/payment_viewing.feature [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+    And the store has "UPS" shipping method with "$20.00" fee per unit for "Web-EU" channel
+    And the store has also a payment method "Bank transfer" with a code "transfer"
+
+
+  @ui
+  Scenario: Browsing payments tab on panel admin
+    Given I am logged in as an administrator
+    And store has primary and secondary order
+    And I am on "/admin/payments/"
+    Then I should see 1 payments
```
Comment (senghe) [inline features/admin/payment_viewing.feature:24]: Where is the information, it's not the payment for primary order?

File: features/admin/shipment_viewing.feature [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+    And the store has "UPS" shipping method with "$20.00" fee per unit for "Web-EU" channel
+    And the store has also a payment method "Bank transfer" with a code "transfer"
+
+
+  @ui
+  Scenario: Browsing shipments tab on panel admin
+    Given I am logged in as an administrator
+    And store has primary and secondary order
+    And I am on "/admin/shipments/"
+    Then I should see 1 shipments
```
Comment (senghe) [inline features/admin/shipment_viewing.feature:24]: Where is the information, it's not the shipment for primary order?

File: features/admin/customer_orders.feature [hunk 17]
```diff
@@ -0,0 +1,26 @@
+@hiding_primary_orders_in_customer_tab
```
Comment (senghe) [inline features/admin/customer_orders.feature:1]: In all behat scenarios I don't see any information (given?), how many orders we have, and which is secondary, and which is primary. When non-tech person reads the .feature file, they should be able to analyse the scenario. Currently I can't.

**Review by senghe — CHANGES_REQUESTED** [review]

File: src/Repository/OrderRepository.php (hunk 1 above)
Comment (MacBalc) [reply src/Repository/OrderRepository.php:84]: Methods below line 84 overwrite methods that are used to calculate dashboard statistics.

File: src/Repository/OrderRepository.php (hunk 2 above)
Comment (MacBalc) [reply src/Repository/OrderRepository.php:77]: Test added

File: src/Repository/OrderRepository.php (hunk 3 above)
Comment (MacBalc) [reply src/Repository/OrderRepository.php:34]: Test added

File: features/admin/customer_orders.feature (hunk 4 above)
Comment (MacBalc) [reply features/admin/customer_orders.feature]: Change applied

File: features/shop/order.feature (hunk 5 above)
Comment (MacBalc) [reply features/shop/order.feature]: I have changed logic to make sure that displayed orders are not primary

File: src/Repository/ShipmentRepository.php (hunk 13 above)
Comment (MacBalc) [reply src/Repository/ShipmentRepository.php]: Name changed

File: features/admin/customer_orders.feature (hunk 14 above)
Comment (MacBalc) [reply features/admin/customer_orders.feature:26]: I have changed logic to make sure that displayed orders are not primary

File: features/admin/payment_viewing.feature (hunk 15 above)
Comment (MacBalc) [reply features/admin/payment_viewing.feature]: I have changed logic to make sure that displayed orders are not primary

File: features/admin/shipment_viewing.feature (hunk 16 above)
Comment (MacBalc) [reply features/admin/shipment_viewing.feature]: I have changed logic to make sure that displayed orders are not primary

File: features/admin/customer_orders.feature (hunk 17 above)
Comment (MacBalc) [reply features/admin/customer_orders.feature:1]: I have changed logic to make sure that displayed orders are not primary

File: src/Repository/OrderRepository.php (hunk 8 above)
Comment (MacBalc) [reply src/Repository/OrderRepository.php:176]: Methods below line 84 overwrite methods that are used to calculate dashboard statistics.

File: src/Repository/OrderRepository.php (hunk 9 above)
Comment (MacBalc) [reply src/Repository/OrderRepository.php:177]: Comment added

File: src/Repository/PaymentRepository.php (hunk 10 above)
Comment (MacBalc) [reply src/Repository/PaymentRepository.php]: Name changed

File: features/admin/payment_viewing.feature (hunk 11 above)
Comment (MacBalc) [reply features/admin/payment_viewing.feature]: Change applied

File: spec/NumberAssigner/OrderNumberAssignerSpec.php (hunk 12 above)
Comment (MacBalc) [reply spec/NumberAssigner/OrderNumberAssignerSpec.php]: Import added

**Pull request merged**